test/unit/matrix/test_copy.cpp

```diff
@@ -135,15 +135,23 @@ TYPED_TEST(MatrixCopyTest, FullMatrixGPU) {
 #endif
 
 struct SubMatrixCopyConfig {
-  GlobalElementSize full_in;
-  GlobalElementSize full_out;
+  const GlobalElementSize full_in;
```

Comment:
Why changes to test_copy?

Reply:
The only real change is that now `SubMatrixCopyConfig` has two member functions that return a `SubMatrixSpec` that can be used with the refactored `mixValues` and `subValues` functions. The rest is just adding some more const-correctness that I realised it was missing.